### PR TITLE
[Communication][Email] Adding tests.yml file

### PR DESCRIPTION
**Pull request by yogeshmo**
# Description
- Adding a tests file for the live test pipeline

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by weshaggard — APPROVED** [review]
Looks good once the test pipeline is passing.

**Review by vcolin7 — APPROVED** [review]

File: sdk/communication/azure-communication-email/src/test/java/com/azure/communication/email/EmailTestBase.java [hunk 1]
```diff
@@ -71,4 +76,15 @@ private static TestMode initializeTestMode() {
...
         }
     }
+
+    static Stream<Arguments> getTestParameters() {
+        // When this issues is closed, the newer version of junit will have better support for cartesian product of
+        // arguments - https://github.com/junit-team/junit5/issues/1427
+        List<Arguments> argumentsList = new ArrayList<>();
+
+        getHttpClients()
+            .forEach(httpClient -> Arguments.of(httpClient));
```
Comment (srnagar) [inline sdk/communication/azure-communication-email/src/test/java/com/azure/communication/email/EmailTestBase.java:86]: The client is not added to the `argumentsList` here. So, the argumentsList is always empty.

**Pull request merged**